source/fe/mapping_cartesian.cc

```diff
@@ -37,6 +37,32 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+DeclExceptionMsg(
+  ExcCellNotCartesian,
+  "You are using MappingCartesian, but the incoming cell is not Cartesian.");
+
+
+
+/**
+ * Return whether the incoming cell is a Cartesian. This is determined by
+ * checking if the smallest BoundingBox that encloses the cell has the same
+ * vertices as the cell itself.
+ */
+template <class CellType>
+bool
+is_cartesian(const CellType &cell)
+{
+  const auto bounding_box = cell->bounding_box();
+  for (unsigned int v : cell->vertex_indices())
+    {
+      const double tolerance = 1e-14;
```

Comment:
I think it would be better to use relative tolerances:
```suggestion
      const double tolerance = 1e-14 * cell->diameter();
```
(I note that this call is somewhat expensive, but this is only used for assertions anyway so we can afford it.)